src/spec/dsl.cr

```diff
@@ -109,11 +109,27 @@ module Spec
     lines << line
   end
 
+  @@split_filter : NamedTuple(remainder: Int32, quotient: Int32)? = nil
+
+  def self.add_split_filter(filter)
+    if filter
+      r, m = filter.split('%').map &.to_i
+      @@split_filter = {remainder: r, quotient: m}
+    else
+      @@split_filter = nil
+    end
+  end
+
   # :nodoc:
   def self.matches?(description, file, line, end_line = line)
     spec_pattern = @@pattern
     spec_line = @@line
     locations = @@locations
+    split_filter = @@split_filter
+
+    if split_filter && line % split_filter[:quotient] != split_filter[:remainder]
```

Comment:
The way I thought about `SPLIT_SPEC`, each time you do `it` a counter would get incremented, and it's this counter you would check against.

By using just the line number it might happen that more specs are in lines 1 modulo 4, for example.

But if it works then it's a good start :-)

Reply:
Yup, that's another way to filter them. The distribution of the std_spec at least was good enough.

```
$ make std_spec
7744 examples, 0 failures, 0 errors, 8 pending

$ SPEC_SPLIT="0%4" make std_spec
1958 examples, 0 failures, 0 errors, 2 pending

$ SPEC_SPLIT="1%4" make std_spec
1856 examples, 0 failures, 0 errors, 2 pending

$ SPEC_SPLIT="2%4" make std_spec
1958 examples, 0 failures, 0 errors, 2 pending

$ SPEC_SPLIT="3%4" make std_spec
1972 examples, 0 failures, 0 errors, 2 pending
```